src/gs-updates-page.c

```diff
@@ -783,6 +784,9 @@ _perform_update_cb (GsPluginLoader *plugin_loader, GAsyncResult *res, gpointer u
 		return;
 	}
 
+	now = g_date_time_new_now_local ();
+	g_settings_set (helper->self->settings, "install-timestamp", "x", g_date_time_to_unix (now));
+
```

Comment:
If you are going to call this from many places, then I think it's worth that you abstract it in a function (probably in `gs-utils`).

Reply:
Ah, I looked for a utils file but couldn't find it. I realized it I was looking in src/ only but there is one in lib/